math/vecops/inc/ROOT/RVec.hxx

```diff
@@ -667,6 +667,27 @@ void swap(RVec<T> &lhs, RVec<T> &rhs)
    lhs.swap(rhs);
 }
 
+/// Return indices that sort the vector
+template <typename T>
+RVec<size_t> Argsort(const RVec<T> &v)
+{
+   using size_type = typename std::vector<T>::size_type;
+   RVec<size_type> i(v.size());
+   std::iota(i.begin(), i.end(), 0);
+   std::sort(i.begin(), i.end(), [&v](size_type i1, size_type i2) { return v[i1] < v[i2]; });
+   return i;
+}
+
+/// Return elements of a vector at given indices
+template <typename T>
+RVec<T> ByIndices(const RVec<T> &v, const RVec<typename std::vector<T>::size_type> &i)
+{
+   RVec<T> r(i.size());
+   for (unsigned int k = 0; k < i.size(); k++)
+      r[k] = v[i[k]];
```

Comment:
Why not a range-based for loop, as in [my post](https://root-forum.cern.ch/t/sort-struct-of-array-format-with-rvec/29485/5)?

Reply:
I don't know, does it make any difference performance wise? But I agree that it looks much nicer.